src/OpenTelemetry/Metrics/MetricAggregators/HistogramMetricAggregator.cs

```diff
@@ -62,11 +121,38 @@ internal HistogramMetricAggregator(string name, string description, string unit,
         public void Update<T>(T value)
             where T : struct
         {
-            // TODO: Implement Histogram!
+            // promote value to be a double
+
+            double val;
+            if (typeof(T) == typeof(long))
+            {
+                val = (long)(object)value;
+            }
+            else if (typeof(T) == typeof(double))
+            {
+                val = (double)(object)value;
+            }
+            else
+            {
+                val = (double)(object)value;
```

Comment:
trust the SDK to send only long or double, anything else - ignore/log/throw.

Reply:
Sure